### PR TITLE
FEAT: move json serialization of EggMetadata from edm.

### LINKED ISSUE #213: Move EggMetadata json serialization from edm
See enthought/edm#719

### PR DESCRIPTION
**Pull request by cournape**
Fix #213 

@johntyree

### CODE REVIEW

File: okonomiyaki/file_formats/_egg_info.py [hunk 1]
```diff
@@ -1015,6 +1054,28 @@ def dump(self, path):
...
+    def to_json_dict(self):
+        if self.platform is None:
+            epd_platform = None
+        else:
+            epd_platform = six.text_type(self.platform)
+
+        return {
+            _JSON_METADATA_VERSION: six.text_type(self.metadata_version),
+            _JSON__RAW_NAME: self._raw_name,
+            _JSON_NAME: self.name,
```
Comment (johntyree) [inline okonomiyaki/file_formats/_egg_info.py]: Why do we serialize this? We do not deserialize it.

Comment (noraderam) [inline okonomiyaki/file_formats/_egg_info.py]: Similar question here: https://github.com/enthought/edm/pull/719/files#r64397216

Comment (johntyree) [inline okonomiyaki/file_formats/_egg_info.py]: I see. Although, that still doesn't answer my question really. Is it something we need or not? The comment there suggests not and here it clearly isn't used at all anyway.

Comment (cournape) [inline okonomiyaki/file_formats/_egg_info.py]: the name vs so-called raw name distinction is needed for various compatibility reasons. But here, `EggMetadata` is handling it for us, with `name` being deduced from the `raw_name`.

**Pull request merged**